proposed/0006-Launcher.md

```diff
@@ -0,0 +1,192 @@
+
+<4> along with [safe_vault](https://github.com/maidsafe/safe_vault) will manage the mapping and de-mapping of crypto and ownership keys for the App (if the App requires to mutate the network on the user's behalf)
+
+## Expected outcome
+
+SAFE-Launcher
+
+<1> will allow user to create an account and/or log in the SAFE-Network.
+
+<2> will authenticate a user installed App to access SAFE-Network on the user's behalf.
+
+<3> will manage metadata related to Apps to give uniformity in experience when shifting from one machine to another - eg., if App `A` is installed in machine 1 then when the user logs into machine 2 using his SAFE-Account via Launcher, he will be presented with a union of all the Apps that were installed on all the machines which access the SAFE-Network on his behalf.
+
+<4> along with [safe_vault](https://github.com/maidsafe/safe_vault) will manage the mapping and de-mapping of crypto and ownership keys for the App (if the App requires to mutate the network on the user's behalf)
+
+# Detailed design
+
+## User's Login Session Packet (for reference)
+This is only to provide a context to the references to it below. This might change in future without affecting this RFC (ie., only a small portion of this is actually relevant for this RFC).
+```
+Account {
+    an_maid,
+    maid,
+    public_maid,
+    an_ampid,
+    mpid,
+    public_mpid,
+    Option<USER’S-PRIVATE-ROOT-DIRECTORY-ID>, // This is easily accessible to the user (ie., mounted as a drive etc.).
+    Option<MAIDSAFE-SPEICIFIC-CONFIG-ROOT>   // This is accessible only if specifically asked and come with a warning to not directly modify it.
+}
+```
+
+## Start Flow
+
+**step 0:** Start Launcher.
+
+**step 1:** Enter Credentials to either create an account or to log into a previously created one.
+
+**step 2:** If it was a log in, Launcher Fetches and decodes User-Session-Packet (USP).
+
+**step 3:** Launcher Fetches Maidsafe Specific Configuration Private Root Directory ID (See Session Packet description) - if not present Launcher will Create it (via [safe_nfs](https://github.com/maidsafe/safe_nfs) crate)
+
+**step 4:** Launcher Reads the special Directory reserved for it. (See Session Packet Description) - if not present Launcher will Create it (via [safe_nfs](https://github.com/maidsafe/safe_nfs) crate)
+
+**step 5:** Launcher Fetches User’s Private root Dir - if not present Launcher will Create it (via [safe_nfs](https://github.com/maidsafe/safe_nfs) crate)
+
+**step 6:** Launcher Checks for special Directory named `SAFEDrive` inside the user’s root directory - if not present Launcher will Create it (via [safe_nfs](https://github.com/maidsafe/safe_nfs) crate)
```

Comment:
I think this would be better as `SAFE Drive` (or if we don't want to handle spaces, then `safe_drive`).  Same for other file/folder names mentioned below - e.g. `LauncherReservedDirectory` -> `Launcher Reserved Directory` and `DnsReservedDirectory` -> `DNS Reserved Directory`.